Add browser support to README

dbrugger opened the pull request:
#### Changelog

**New**

- add list of supported browsers to README

asfordmatt commented:
I'd recommend saying the latest version of those browsers. 
Also do we need to include ESR versions of Firefox?

dbrugger commented:
Thank you @asfordmatt for your input on this! I've added mention of _latest_ and removed ESR for now so we can start with a slim baseline but happy to get further input!
